### DISCUSSION TITLE
How to use openpnm to simulate single-phase infiltration after extraction of pores using the snow2 algorithm

### DISCUSSION THREAD

**Discussion by Clifford-CAS** [PMEAL/OpenPNM#2687] [Q&A]
How to use openpnm to simulate single-phase infiltration after extraction of pores using the snow2 algorithm
![image](https://user-images.githubusercontent.com/105999156/217772446-42a7cddf-2f0c-4962-9b9b-47831e5331de.png)

**Comment by ma-sadeghi** [comment]
Hi @wangjiayun258138. You need to be more specific so developers can help you out.

**Reply by jgostick** [reply]
You say single phase, but also mention infiltration.  If you are looking for invasion of a fluid, see [this example](https://openpnm.org/examples/tutorials/09_simulating_invasion.html)

**Reply by Clifford-CAS** [reply]
Thank you for the perfect explanation!

**Reply by jgostick** [reply]
The examples are now in OpenPNM V3 syntax, but the code you pasted above is V2.  The switch is not too difficult, but there is an explanation [here](http://pmeal.com/posts/2022-10-10-notebook-post/)
